### PR TITLE
fix(Guidebanner): allow adding 1rem space on the left

### PR DESCRIPTION
**Pull request by paul-balchin-ibm**
Contributes to #3903

New prop `withLeftGutter`

#### How did you test and verify your work?

- [x] Storybook review
- [x] Design reviewers: Cameron Calder, Shelley Dignum, Bridget Ndege.

### CODE REVIEW

File: packages/ibm-products/src/components/Guidebanner/Guidebanner.js [hunk 1]
```diff
@@ -69,8 +71,9 @@ export let Guidebanner = React.forwardRef(
         className={cx(
           blockClass,
           className,
-          [collapsible ? `${blockClass}__collapsible` : null],
-          [isCollapsed ? `${blockClass}__collapsible-collapsed` : null]
+          collapsible && `${blockClass}__collapsible`,
+          isCollapsed && `${blockClass}__collapsible-collapsed`,
```
Comment (paul-balchin-ibm) [inline packages/ibm-products/src/components/Guidebanner/Guidebanner.js:75]: Extra: simplified code.

Comment (elycheea) [reply packages/ibm-products/src/components/Guidebanner/Guidebanner.js:75]: Haha, is this pulling from your recent convo with @davidmenendez 😆

File: packages/ibm-products/src/components/Guidebanner/Guidebanner.stories.js
Comment (elycheea) [inline packages/ibm-products/src/components/Guidebanner/Guidebanner.stories.js:1]: If the grid matters here, I actually think it might make more sense to change the layout style from `padded` to `fullscreen` so you can see the alignment using Carbon Devtools.

image

Comment (paul-balchin-ibm) [reply packages/ibm-products/src/components/Guidebanner/Guidebanner.stories.js:1]: A component's only direct connection to the outside world should be limited to changing its props values. It's like an API: "apply setting to component" and "listen to changes from component" and nothing else. Trying to get a component to react to external features like page or grid settings can result too easily result in false positives.

Comment (elycheea) [reply packages/ibm-products/src/components/Guidebanner/Guidebanner.stories.js:1]: Oh. Let me rephrase — change the _Storybook_ layout setting. My suggestion is not a change to the component itself. I think this is L34 in Stories file but will give a more accurate representation of how it looks on the page. (PageHeader and Datagrid sometimes use this setting instead of the padded style in Storybook.)

image

Comment (paul-balchin-ibm) [reply packages/ibm-products/src/components/Guidebanner/Guidebanner.stories.js:1]: OH! I see what you mean. Sure, I can change that.

**Review by elycheea — APPROVED** [review]

File: packages/ibm-products/src/components/Guidebanner/Guidebanner.js (hunk 1 above)
Comment (paul-balchin-ibm) [reply packages/ibm-products/src/components/Guidebanner/Guidebanner.js:75]: Actually, I had completely missed his comment while making these changes. :)
